payments-core/src/main/java/com/stripe/android/googlepaylauncher/GooglePayRepository.kt

```diff
@@ -66,32 +62,35 @@ internal class DefaultGooglePayRepository(
      * See [Google Pay API docs](https://developers.google.com/android/reference/com/google/android/gms/wallet/PaymentsClient#isReadyToPay(com.google.android.gms.wallet.IsReadyToPayRequest))
      * for more details.
      */
-    override fun isReady(): Flow<Boolean> {
-        val isReadyState = MutableStateFlow<Boolean?>(null)
+    override fun isReady(): Flow<Boolean> = flow {
+        emit(isReadyAsync())
+    }
 
-        runCatching {
-            val request = IsReadyToPayRequest.fromJson(
+    private suspend fun isReadyAsync(): Boolean {
+        val request = runCatching {
+            IsReadyToPayRequest.fromJson(
                 googlePayJsonFactory.createIsReadyToPayRequest(
                     billingAddressParameters = billingAddressParameters,
                     existingPaymentMethodRequired = existingPaymentMethodRequired,
                     allowCreditCards = allowCreditCards
                 ).toString()
             )
+        }.getOrElse {
+            // TODO (samer-stripe): Add unexpected error event here
+            logger.error("Google Pay json parsing failed.", it)
 
-            paymentsClient.isReadyToPay(request)
-                .addOnCompleteListener { task ->
-                    val isReady = runCatching {
-                        task.getResult(ApiException::class.java) == true
-                    }.onFailure {
-                        logger.error("Google Pay check failed.", it)
-                    }.getOrDefault(false)
-                    logger.info("Google Pay ready? $isReady")
-                    isReadyState.value = isReady
-                }
-        }.onFailure {
-            isReadyState.value = false
+            return false
         }
 
-        return isReadyState.filterNotNull()
+        val isReady = runCatching {
+            paymentsClient.isReadyToPay(request).await()
+        }.onFailure {
+            // TODO (samer-stripe): Add error event here
+            logger.error("Google Pay check failed.", it)
+        }.getOrDefault(false)
+
+        logger.info("Google Pay ready? $isReady")
+
+        return isReady
```

Comment:
Performs the same logic as `isReady` does in master but this is function is suspendable, returns a `Boolean` value and uses `org.jetbrains.kotlinx:kotlinx-coroutines-play-services`'s `await` coroutine helper rather than a `onCompleteListener`.